### ISSUE TITLE
Allow to specify docker image

### ISSUE THREAD

**Issue by stkw0** [nicolasbock/ebuildtester#150]
Allowing to specify the docker image will allow to use docker images for other architectures and test packages with qemu. Or maybe it will be better to specify the architecture ¿?

**Comment by nicolasbock** [comment]
This feature was add in nicolasbock/ebuildtester#146. I just realized though that I didn't document it in the README :/

I started using this flag because the official docker image hasn't been updated in the last 6 months (https://hub.docker.com/r/gentoo/stage3-amd64). In terms of architectures, that's an interesting angle. I hadn't considered that. Would that work out of the box? Or would we have to change something in the docker setup?

**Comment by stkw0** [comment]
Seems that "ebuildtester --help" neither prints that option

**Comment by stkw0** [comment]
Regarding your second question, following the blogs I found on the internet, I executed: `docker run --rm --privileged multiarch/qemu-user-static --reset -p yes ` to configure the binfmt_misc thing and afterwards it should work of the box. When I have time I will try this hidden feature of ebuildtester to see how it works  xD

**Comment by nicolasbock** [comment]
> Seems that "ebuildtester --help" neither prints that option

It's included in version `v0.1.27`. I will update the README as well to include that option.

**Comment by nicolasbock** [comment]
> Regarding your second question, following the blogs I found on the internet, I executed: `docker run --rm --privileged multiarch/qemu-user-static --reset -p yes ` to configure the binfmt_misc thing and afterwards it should work of the box. When I have time I will try this hidden feature of ebuildtester to see how it works xD

That would be amazing! Thanks!

**Comment by stkw0** [comment]
> > Seems that "ebuildtester --help" neither prints that option
> 
> It's included in version `v0.1.27`. I will update the README as well to include that option.

I am not able to see it

```
❱ ebuildtester --version
v0.1.27

❱ ebuildtester --help 
usage: ebuildtester [-h] [--version] [--atom ATOM [ATOM ...]] [--binhost BINHOST] [--live-ebuild] [--manual] --portage-dir PORTAGE_DIR [--overlay-dir OVERLAY_DIR] [--update] [--install-basic-packages] [--threads N] [--use USE [USE ...]]
                    [--global-use GLOBAL_USE [GLOBAL_USE ...]] [--unmask ATOM] [--unstable] [--gcc-version VER] [--python-single-target PYTHON_SINGLE_TARGET] [--python-targets PYTHON_TARGETS] [--rm] [--pull] [--storage-opt STORAGE_OPT [STORAGE_OPT ...]] [--with-X]
                    [--with-vnc] [--profile {default/linux/amd64/17.1,default/linux/amd64/17.1/systemd}]

A dockerized approach to test a Gentoo package within a clean stage3.

optional arguments:
  -h, --help            show this help message and exit
  --version             show program's version number and exit
  --atom ATOM [ATOM ...]
                        The package atom(s) to install
  --binhost BINHOST     Binhost URI
  --live-ebuild         Unmask the live ebuild of the atom
  --manual              Install package manually
  --portage-dir PORTAGE_DIR
                        The local portage directory
  --overlay-dir OVERLAY_DIR
                        Add overlay dir (can be used multiple times)
  --update              Update container before installing atom
  --install-basic-packages
                        Install basic packages after container starts
  --threads N           Use N (default 8) threads to build packages
  --use USE [USE ...]   The use flags for the atom
  --global-use GLOBAL_USE [GLOBAL_USE ...]
                        Set global USE flag
  --unmask ATOM         Unmask atom (can be used multiple times)
  --unstable            Globally 'unstable' system, i.e. ~amd64
  --gcc-version VER     Use gcc version VER
  --python-single-target PYTHON_SINGLE_TARGET
                        Specify a PYTHON_SINGLE_TARGET
  --python-targets PYTHON_TARGETS
                        Specify a PYTHON_TARGETS
  --rm                  Remove container after session is done
  --pull                Download latest 'gentoo/stage3-amd64' docker image
  --storage-opt STORAGE_OPT [STORAGE_OPT ...]
                        Storage driver options for all volumes (same as Docker param)
  --with-X              Globally enable the X USE flag
  --with-vnc            Install VNC server to test graphical applications
  --profile {default/linux/amd64/17.1,default/linux/amd64/17.1/systemd}
                        The profile to use
```

**Comment by nicolasbock** [comment]
I just realized that the image was renamed to https://hub.docker.com/r/gentoo/stage3

**Comment by stkw0** [comment]
> > > Seems that "ebuildtester --help" neither prints that option
> > 
> > 
> > It's included in version `v0.1.27`. I will update the README as well to include that option.
> 
> I am not able to see it
> 
> ```
> ❱ ebuildtester --version
> v0.1.27
> 
> ❱ ebuildtester --help 
> usage: ebuildtester [-h] [--version] [--atom ATOM [ATOM ...]] [--binhost BINHOST] [--live-ebuild] [--manual] --portage-dir PORTAGE_DIR [--overlay-dir OVERLAY_DIR] [--update] [--install-basic-packages] [--threads N] [--use USE [USE ...]]
>                     [--global-use GLOBAL_USE [GLOBAL_USE ...]] [--unmask ATOM] [--unstable] [--gcc-version VER] [--python-single-target PYTHON_SINGLE_TARGET] [--python-targets PYTHON_TARGETS] [--rm] [--pull] [--storage-opt STORAGE_OPT [STORAGE_OPT ...]] [--with-X]
>                     [--with-vnc] [--profile {default/linux/amd64/17.1,default/linux/amd64/17.1/systemd}]
> 
> A dockerized approach to test a Gentoo package within a clean stage3.
> 
> optional arguments:
>   -h, --help            show this help message and exit
>   --version             show program's version number and exit
>   --atom ATOM [ATOM ...]
>                         The package atom(s) to install
>   --binhost BINHOST     Binhost URI
>   --live-ebuild         Unmask the live ebuild of the atom
>   --manual              Install package manually
>   --portage-dir PORTAGE_DIR
>                         The local portage directory
>   --overlay-dir OVERLAY_DIR
>                         Add overlay dir (can be used multiple times)
>   --update              Update container before installing atom
>   --install-basic-packages
>                         Install basic packages after container starts
>   --threads N           Use N (default 8) threads to build packages
>   --use USE [USE ...]   The use flags for the atom
>   --global-use GLOBAL_USE [GLOBAL_USE ...]
>                         Set global USE flag
>   --unmask ATOM         Unmask atom (can be used multiple times)
>   --unstable            Globally 'unstable' system, i.e. ~amd64
>   --gcc-version VER     Use gcc version VER
>   --python-single-target PYTHON_SINGLE_TARGET
>                         Specify a PYTHON_SINGLE_TARGET
>   --python-targets PYTHON_TARGETS
>                         Specify a PYTHON_TARGETS
>   --rm                  Remove container after session is done
>   --pull                Download latest 'gentoo/stage3-amd64' docker image
>   --storage-opt STORAGE_OPT [STORAGE_OPT ...]
>                         Storage driver options for all volumes (same as Docker param)
>   --with-X              Globally enable the X USE flag
>   --with-vnc            Install VNC server to test graphical applications
>   --profile {default/linux/amd64/17.1,default/linux/amd64/17.1/systemd}
>                         The profile to use
> ```

I installed ebuildtester from portage but seems some imports where made from my user install, removing it fixed the issue

**Comment by nicolasbock** [comment]
Oh cool. I was getting worried :)
